lightning-graph-sync/src/processing.rs

```diff
@@ -0,0 +1,550 @@
+use std::cmp::max;
+use std::io;
+use std::io::Read;
+
+use bitcoin::secp256k1::PublicKey;
+use bitcoin::BlockHash;
+
+use lightning::ln::msgs::{
+    DecodeError, ErrorAction, LightningError, OptionalField, UnsignedChannelUpdate,
+};
+use lightning::routing::network_graph;
+use lightning::util::ser::{BigSize, Readable};
+
+use crate::error::GraphSyncError;
+use crate::GraphSyncError::ProcessingError;
+
+/// The purpose of this prefix is to identify the serialization format, should other rapid gossip
+/// sync formats arise in the future.
+///
+/// The fourth byte is the protocol version in case our format gets updated.
+const GOSSIP_PREFIX: [u8; 4] = [76, 68, 75, 1];
+
+/// Maximum vector allocation capacity for distinct node IDs. This constraint is necessary to
+/// avoid malicious updates being able to trigger excessive memory allocation.
+const MAX_INITIAL_NODE_ID_VECTOR_CAPACITY: u32 = 50_000;
+
+/// Update network graph from binary data
+///
+/// `network_graph`: network graph to be updated
+///
+/// `update_data`: `&[u8]` binary stream that comprises the update data
+pub fn update_network_graph(
+    network_graph: &network_graph::NetworkGraph,
```

Comment:
Looks like the tabs in this file was converted to spaces.

Reply:
Ah I ran rustfmt for idiomatic 100-column wrapping. Will undo.